src/logger.coffee

```diff
@@ -5,114 +5,237 @@ es = require 'event-stream'
 Lock = require 'rwlock'
 { checkTruthy } = require './lib/validation'
 
-LOG_PUBLISH_INTERVAL = 110
+class NativeLoggerBackend
+	constructor: ->
+		@logPublishInterval = 1000
+		@maxLogLinesPerBatch = 10
+		@maxQueueSize = 100
+		@maxLineLength = 300
+		@publishQueue = []
+		@intervalHandle = null
+		@publishEnabled = false
+		@offlineMode = false
+		@shouldStop = false
+
+	publishLoop: =>
+		startTime = process.hrtime()
+		Promise.try =>
+			return if @offlineMode or !@publishEnabled or @publishQueue.length is 0
+			currentBatch = _.slice(@publishQueue, 0, @maxLogLinesPerBatch)
```

Comment:
NIT: You can do both things in one step with `currentBatch = @publishQueue.splice(0, @maxLogLinesPerBatch)` and you'll be creating less arrays too

Reply:
Nice one, done